### PR TITLE
New navigation

### LINKED ISSUE #1049: Implement new navigation design
Spec: https://github.com/ubuntudesign/vanilla-design/tree/master/Navigation

### PR DESCRIPTION
**Pull request by anthonydillon**
## Done
Built out the two new navigation styles referencing the spec. Removed the fixed-width and full-width documentation and combined them into a note on the navigation documentation. Created a light and dark navigation example with documentation.

## QA
- Pull code
- Run `gulp test` and check there are no lint errors
- Run `gulp jekyll`
- Go to http://127.0.0.1:4000/vanilla-framework/examples/patterns/navigation/light/ and http://127.0.0.1:4000/vanilla-framework/examples/patterns/navigation/dark/
- Check that they both match the [design specs](https://github.com/ubuntudesign/vanilla-design/tree/master/Navigation)

## Details
Fixes: https://github.com/vanilla-framework/vanilla-framework/issues/1049

### CODE REVIEW

File: examples/patterns/navigation/light.html [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+<header id="navigation" class="p-navigation--light" role="banner">
+  <div class="row">
+    <div class="p-navigation__logo">
+      <a class="p-navigation__link" href="#">
+        Vanilla
+      </a>
+    </div>
+    <a href="#navigation" class="p-navigation__toggle--open" title="menu">Menu</a>
+    <a href="#navigation-closed" class="p-navigation__toggle--close" title="close menu">Menu</a>
```
Comment (richmccartney) [inline examples/patterns/navigation/light.html]: - For a11y both links should be a button element not a link. We may need to rethink how the toggle works in the future? 

- Requires aria tags as well for elements it controls and visibility.

File: scss/_patterns_navigation.scss [hunk 2]
```diff
@@ -100,31 +127,37 @@
       }
 
       .p-navigation__link {
```
Comment (richmccartney) [inline scss/_patterns_navigation.scss]: In some websites we also have external links in the header, should we add a BEM extension class to add this style?

File: examples/patterns/navigation/light.html [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    <nav class="p-navigation__nav">
```
Comment (richmccartney) [inline examples/patterns/navigation/light.html]: We should be looking to improve the a11y for the navigation, we fail accessiblity if we use two toggle links. I suggest we change these to ```button``` elements and use aria tags to signify whether it's expanded or not.

File: scss/_patterns_navigation.scss (hunk 2 above)
Comment (richmccartney) [inline scss/_patterns_navigation.scss]: - Should we have a ```external``` class as we have this pattern on some websites already?
- When hovering on a ```.p-navigation__link``` the cursor stays at default until you are over the text.

File: examples/patterns/navigation/light.html (hunk 1 above)
Comment (anthonydillon) [inline examples/patterns/navigation/light.html]: I agree but this needs some thinking as we need javascript for that. Happy to pick it up as part of https://github.com/vanilla-framework/vanilla-framework/issues/971.

File: scss/_patterns_navigation.scss (hunk 2 above)
Comment (anthonydillon) [inline scss/_patterns_navigation.scss]: Link external works although blue which is another issue:
navigation - dark - vanilla framework - examples

Comment (anthonydillon) [inline scss/_patterns_navigation.scss]: I am aware of the padding issue. I need to change the mark up for that so will be a bigger task.

**Comment by anthonydillon** [comment]
@richmccartney thanks for the review. They are a little of the scope for this PR and are issues which I'll get to soon. Please feel free to review again.

File: scss/_patterns_navigation.scss (hunk 2 above)
Comment (richmccartney) [inline scss/_patterns_navigation.scss]: @anthonydillon Ah okay, well maybe I'll make an issue for the external link and to get an appropriate asset for it.

**Comment by richmccartney** [comment]
@anthonydillon All looks good to me +1 aside from the talking points raised in this PR.

File: scss/_patterns_navigation.scss (hunk 2 above)
Comment (anthonydillon) [inline scss/_patterns_navigation.scss]: Yes please do create an issue. We need to external link to use the currentColor.

**Comment by anthonydillon** [comment]
Thanks @richmccartney sorry I didn't address the talking points as I agree with them all but I'll get to them as soon as I can.

File: scss/_patterns_navigation.scss (hunk 2 above)
Comment (yaili) [inline scss/_patterns_navigation.scss]: Should the padding be in the `<a>` instead of the `<li>`?

**Pull request merged**